cpp-ch/local-engine/Shuffle/PartitionWriter.cpp

```diff
@@ -228,11 +228,11 @@ CelebornPartitionWriter::CelebornPartitionWriter(CachedShuffleWriter * shuffleWr
 void CelebornPartitionWriter::evictPartitions(bool for_memory_spill)
 {
     auto spill_to_celeborn = [this]() -> void {
-        size_t partition_id = 0;
         Stopwatch serialization_time_watch;
         serialization_time_watch.start();
-        for (auto & partition : partition_buffer)
+        for (size_t partition_id = 0; partition_id < shuffle_writer->options.partition_nums; ++partition_id)
```

Comment:
why not use `partition_buffer.size()`, at least add a assert `assert(partition_buffer.size() == shuffle_writer->options.partition_nums)`